src/utilities.jl

```diff
@@ -95,7 +97,7 @@ Collect and return all methods of function `f` matching signature `sig`.
 This is similar to `methods(f, sig)`, but handles type signatures found in `DocStr` objects
 more consistently that `methods`.
 """
-getmethods(f, sig) = getmethods!(Method[], f, sig)
+getmethods(f, sig) = unique(getmethods!(Method[], f, sig))
```

Comment:
The only real use case wrapped it in `Set` straight away anyway, and tests were checking for a specific number of methods. With `UnionAll` types, it's easy to get multiple matches.